docs/index.md

```diff
@@ -0,0 +1,291 @@
+# Go Datadog Library
```

Comment:
Is there a practical reason why it collapsed in one long file now?

Reply:
Yes and I should have stated so, there is so little content that I do not think the reader benefits from having to navigate across multiple file.

TechDocs will generate and show a table of content for each file and display it to the right of the content.

I can revert to a multi file setup if that is preferred.

Reply:
It's fine, if you like it more we can do more of it.

Reply:
It is a joint effort 😄